### DISCUSSION TITLE
Should `_typeshed.Incomplete` be special cased as `Unknown`?

### DISCUSSION THREAD

**Discussion by Avasam** [microsoft/pyright#4471] [General]
typeshed tries to leave functions, variables and parameters unannotated when it can, which allows pyright to pick those up as `Unknown`. Unfortunately this is not always possible. In which case typeshed uses the alias `Incomplete`, which is semantically equivalent. But this ends up hiding a lot of "Unknown/partially Unknown" errors.

**Comment by erictraut** [comment]
`Incomplete` isn't an official symbol in the type system or in the Python runtime, and it's not documented in any PEPs. I'd therefore be reluctant to give it any special-case meaning in pyright. It's defined in `typeshed.pyi` as a type alias for `Any`, so that's how pyright will treat it.
